pkg/engine/events.go

```diff
@@ -99,6 +102,15 @@ const (
 	PolicyViolationEvent    EventType = "policy-violation"
 	PolicyRemediationEvent  EventType = "policy-remediation"
 	PolicyLoadEvent         EventType = "policy-load"
+	DownloadProgressEvent   EventType = "download-progress"
```

Comment:
I think when adding a new even type we need to make `pulumi-service` aware of this first, otherwise it will panic when trying to display the event. See https://github.com/pulumi/pulumi-service/commit/015005ab555f8bb6cb6c50c32458545867eadd57

Reply:
Yeah I'd love to see the PR describe explicitly whether these events are going to Cloud and the impact of that. I'm assume they don't need to leave the CLI?

Reply:
Yea I believe that the introduction of `IsEphemeral` and its use in `httpstate` mean that these events *are not* being sent to the service (and thus we don't need to make changes there). I did investigate improving the rendering of this sort of thing on the service side, but their fork of the display code is sufficiently different that it'd take a bit of time and so I'm leaving that out of scope for now. If we do want to pick up that work though, we can mark these events as not ephemeral and send them over for the service to improve its display also.